### PR TITLE
Firestore/cleanup

### PR DESCRIPTION
**Pull request by quiro91**
## Problem

After moving from the RTD to Firestore there's a bunch of stuff we need to clean up a bit

## Solution

Clean it up. This is the first of two PRs to address this issue

### Paired with 

Nobody

### CODE REVIEW

File: app/src/main/java/net/squanchy/eventdetails/EventDetailsService.kt [hunk 1]
```diff
@@ -6,8 +6,8 @@ import io.reactivex.Completable
 import io.reactivex.Observable
 import io.reactivex.Single
 import net.squanchy.schedule.domain.view.Event
-import net.squanchy.service.firebase.FirebaseAuthService
-import net.squanchy.service.repository.EventRepository
+import net.squanchy.service.firestore.FirebaseAuthService
```
Comment (fourlastor) [inline app/src/main/java/net/squanchy/eventdetails/EventDetailsService.kt]: this isn't firestore specific

File: app/src/main/java/net/squanchy/eventdetails/EventDetailsService.kt [hunk 2]
```diff
@@ -6,8 +6,8 @@ import io.reactivex.Completable
...
+import net.squanchy.service.repository.firestore.EventRepository
```
Comment (fourlastor) [inline app/src/main/java/net/squanchy/eventdetails/EventDetailsService.kt]: this isn't firestore specific

File: app/src/main/java/net/squanchy/home/HomeComponent.kt [hunk 3]
```diff
@@ -8,6 +9,14 @@ import net.squanchy.navigation.Navigator
 import net.squanchy.support.injection.CurrentTimeModule
 
 import dagger.Component
+import net.squanchy.injection.ActivityContextModule
+import net.squanchy.injection.applicationComponent
+
+internal fun homeComponent(activity: AppCompatActivity) =
```
Comment (fourlastor) [inline app/src/main/java/net/squanchy/home/HomeComponent.kt]: let's drop `internal`

File: app/src/main/java/net/squanchy/schedule/ScheduleComponent.kt [hunk 4]
```diff
@@ -10,10 +10,18 @@ import net.squanchy.injection.applicationComponent
 import net.squanchy.navigation.NavigationModule
 import net.squanchy.navigation.Navigator
 import net.squanchy.schedule.tracksfilter.TracksFilter
-import net.squanchy.service.repository.TracksRepository
+import net.squanchy.service.repository.firestore.TracksRepository
```
Comment (fourlastor) [inline app/src/main/java/net/squanchy/schedule/ScheduleComponent.kt]: this one isn't firestore specific

File: app/src/main/java/net/squanchy/service/firestore/injection/FirestoreModule.kt [hunk 5]
```diff
@@ -1,20 +1,31 @@
...
+import com.google.firebase.auth.FirebaseAuth
 import com.google.firebase.firestore.FirebaseFirestore
 import dagger.Module
 import dagger.Provides
 import net.squanchy.injection.ApplicationLifecycle
+import net.squanchy.service.firestore.FirebaseAuthService
 import net.squanchy.service.firestore.FirestoreDbService
 
 @Module
 class FirestoreModule {
```
Comment (fourlastor) [inline app/src/main/java/net/squanchy/service/firestore/injection/FirestoreModule.kt]: probably the best thing is to have everything inside a `firebase` package (which is the 3rd party offering the services) and then we'll have `FirestoreDbService` (which is the particular firebase service we're using)

File: app/src/main/java/net/squanchy/service/repository/firestore/FirestoreEventRepository.kt [hunk 6]
```diff
@@ -10,10 +10,20 @@ import net.squanchy.service.firestore.model.conferenceinfo.FirestoreVenue
 import net.squanchy.service.firestore.model.schedule.FirestoreEvent
 import net.squanchy.service.firestore.model.schedule.FirestoreFavorite
 import net.squanchy.service.firestore.toEvent
-import net.squanchy.service.repository.EventRepository
 import net.squanchy.support.lang.Checksum
 import org.joda.time.DateTimeZone
 
+interface EventRepository {
```
Comment (fourlastor) [inline app/src/main/java/net/squanchy/service/repository/firestore/FirestoreEventRepository.kt]: :x: those interface shouldn't be in `firestore`

**Review by fourlastor — CHANGES_REQUESTED** [review]
💯 for the changes, only 2 major points:

- interfaces should stay where they belong, implementation agnostic, not in the firestore package
- firebase is a service provider that offers firestore and rtdb as services, I think the package `firebase` is more appropriate, (firebase firestore, firebase authentication, etc..)

**Comment by quiro91** [comment]
@fourlastor back to review

**Review by fourlastor — APPROVED** [review]

**Pull request merged**